packages/jsii-spec/lib/spec.ts

```diff
@@ -180,10 +180,80 @@ export interface PackageVersion {
 
 /**
  * Key value pairs of documentation nodes.
- * Based on JSDoc.
+ * Based on TSDoc.
  */
 export interface Docs {
-    [key: string]: string;
+    /**
+     * Summary documentation for an API item.
+     *
+     * The first part of the documentation before hitting a `@remarks` tags, or the first
+     * line of the doc comment block if there is no `@remarks` tag.
+     */
+    summary?: string;
+
+    /**
+     * Detailed information about an API item.
+     *
+     * Either the explicitly tagged `@remarks` section, otherwise everything past the
+     * first paragraph if there is no `@remarks` tag.
+     */
+    remarks?: string;
+
+    /**
+     * If present, this block indicates that an API item is no longer supported and may be
+     * removed in a future release.  The `@deprecated` tag must be followed by a sentence
+     * describing the recommended alternative.  Deprecation recursively applies to members
+     * of a container.  For example, if a class is deprecated, then so are all of its members.
+     */
+    deprecated?: string;
+
+    /**
+     * The `@returns` block for this doc comment, or undefined if there is not one.
+     */
+    returns?: string;
+
+    /**
+     * Whether the API item is beta/experimental quality
+     */
+    stability?: Stability;
+
+    /**
+     * Example showing the usage of this API item
+     *
+     * Starts off in running text mode, may switch to code using fenced code
+     * blocks.
+     */
+    example?: string;
+
+    /**
+     * A `@see` link with more information
+     */
+    see?: string;
+
+    /**
+     * Whether this class or interface was intended to be subclassed/implemented by library users.
+     *
+     * Classes intended for subclassing, and interfaces intended to be implemented
+     * by consumers, are held to stricter standards of API compatibility.
+     *
+     * @default false
+     */
+    subclassable?: boolean;
```

Comment:
I am wondering if we should separate all these flags from "docs" into some some other struct associated with each type, but I guess pragmatically we can leave them here.

Reply:
Was thinking the same, ended up with this for now.

Reply:
Maybe also we can expose them as properties in jsii-reflect to improve discoverability...

Reply:
I was originally doing that, but because of the type inheritance tree in `jsii-reflect` it turned into copy/pasting a bunch of similar properties around and then I decided to just put everything on `Docs`. But yeah.